core/src/main/java/org/elasticsearch/index/query/SpanQueryBuilder.java

```diff
@@ -19,8 +19,9 @@
 
 package org.elasticsearch.index.query;
 
-import org.elasticsearch.common.xcontent.ToXContent;
-
-public interface SpanQueryBuilder extends ToXContent {
+/**
+ * Marker interface for a specific type of {@link QueryBuilder} that allows to build span queries
+ */
+public interface SpanQueryBuilder<QB extends SpanQueryBuilder> extends QueryBuilder<QB> {
```

Comment:
Can we also restrict the return type of toQuery() for the queries implementing SpanQueryBuilder to `org.apache.lucene.search.spans.SpanQuery`, because I think that's the main reason for even having this intermediate interface. I think this works, we only need to temporarily reroute all SpanQueryBuilders to the parsers former `parse()` method as long as we havent refactored them. I did something like this in https://github.com/cbuescher/elasticsearch/commit/4ffa1c48c258abc1a3f3a74a47a133ea56a1483d.